org.adempiere.base.process/src/org/compiere/process/CreateFromInvoice.java

```diff
@@ -235,6 +215,20 @@ else if (Value_Date != null)
 				}
 			}
 			
+			ColumnName = "M_InOut_ID";
+			key = ColumnName + "_" + T_Selection_ID;
+			value = selectionValueMap.get(key);
+			int M_InOut_ID = value != null ? ((Number) value).intValue() : 0;
+			if (M_InOut_ID != 0 && (m_inout == null || m_inout.getM_InOut_ID() != M_InOut_ID))
```

Comment:
the correct test for valid id is > 0, not != 0. In other word, it should be M_InOut_ID > 0 instead of M_InOut_ID != 0 (-1 is != 0 but is not a valid id).